brave/brave-core
# Added import of links and text content to ipfs

Resolves https://github.com/brave/brave-browser/issues/14769
Resolves https://github.com/brave/brave-browser/issues/15176

- Added submenu to render context menu
- Added base import worker which implements importing flow for ipfs: add/mkdir/cp apis should be called sequentially
- Added http{s} links importer which downloads a link content to file and uploads it to ipfs
- Added text importer that puts to ipfs a file with passed text, an index for file calculated based on content hash, same text will have same index.
- Launch ipfs daemon if import called and daemon is turned off
- Open of ipfs webui page with imported folder 
- Copy to clipboard of shareable link for uploaded content
- Import calls are proxied through tab helper to have access to chrome/ for opening new tabs when import is finished
- Pre warm request for shareable link
- Added notifications, if import successful it opens shareable link, otherwise brave://ipfs

![image](https://user-images.githubusercontent.com/2965009/114029273-04e1a800-9882-11eb-9891-620903ced74d.png)

![image](https://user-images.githubusercontent.com/2965009/114029170-ed0a2400-9881-11eb-8553-90906e54e269.png)

![image](https://user-images.githubusercontent.com/2965009/114027063-aadfe300-987f-11eb-8191-0729af96c436.png)

## Submitter Checklist:


## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:
- Import different objects on pages as described in ticket
- for each imported object should be opened web page from local node webui in case of success with brave import directory
- for each imported object should be copied shareable link to clipboard which allows to access imported content via public gateway from any other computer
- please pay attention that if you import different selected texts from same page they should be imported all to different text files with indexes
- if you import same selected text it should not add new text file to import folder
- Make sure imported content (link or text) is same as original, try to download file by shareable link it should be same without additional bytes
- If import successful by click on notification should be opened shareable link, otherwise brave://ipfs-internal
